planner/core/task.go

```diff
@@ -146,6 +146,35 @@ func (p *PhysicalIndexJoin) attach2Task(tasks ...task) task {
 	}
 }
 
+func (p *PhysicalIndexMergeJoin) attach2Task(tasks ...task) task {
+	outerTask := finishCopTask(p.ctx, tasks[p.OuterIndex].copy())
+	if p.OuterIndex == 0 {
+		p.SetChildren(outerTask.plan(), p.innerPlan)
+	} else {
+		p.SetChildren(p.innerPlan, outerTask.plan())
+	}
+	p.schema = buildPhysicalJoinSchema(p.JoinType, p)
+	return &rootTask{
+		p:   p,
+		cst: outerTask.cost() + p.getCost(outerTask.count()),
+	}
+}
+
+func (p *PhysicalIndexMergeJoin) getCost(lCnt float64) float64 {
+	if lCnt < 1 {
+		lCnt = 1
+	}
+	cst := lCnt * netWorkFactor
+	batchSize := p.ctx.GetSessionVars().IndexJoinBatchSize
+	sortFactor := 1.0
```

Comment:
The initial value of `sortFactor` should be 0 since if `NeedOuterSort` is false, we will not do sort operation.

Reply:
Range building will cost 1.0 outer rows iteration.